test/integration/gardenlet/seed/seed/seed_test.go

```diff
@@ -333,25 +364,27 @@ var _ = Describe("Seed controller tests", func() {
 					By("Delete Seed")
 					Expect(testClient.Delete(ctx, seed)).To(Succeed())
 
-					By("Verify that the seed system components have been deleted")
-					Eventually(func(g Gomega) []resourcesv1alpha1.ManagedResource {
-						managedResourceList := &resourcesv1alpha1.ManagedResourceList{}
-						g.Expect(testClient.List(ctx, managedResourceList, client.InNamespace(testNamespace.Name))).To(Succeed())
-						return managedResourceList.Items
-					}).Should(BeEmpty())
+					if seedIsGarden {
+						By("Verify that the seed system components have been deleted")
+						Eventually(func(g Gomega) []resourcesv1alpha1.ManagedResource {
+							managedResourceList := &resourcesv1alpha1.ManagedResourceList{}
+							g.Expect(testClient.List(ctx, managedResourceList, client.InNamespace(testNamespace.Name))).To(Succeed())
+							return managedResourceList.Items
+						}).WithTimeout(10 * time.Second).Should(BeEmpty())
+					}
 
 					if !seedIsGarden {
 						By("Verify that gardener-resource-manager has been deleted")
 						Eventually(func(g Gomega) error {
 							deployment := &appsv1.Deployment{ObjectMeta: metav1.ObjectMeta{Name: "gardener-resource-manager", Namespace: testNamespace.Name}}
 							return testClient.Get(ctx, client.ObjectKeyFromObject(deployment), deployment)
-						}).Should(BeNotFoundError())
+						}).WithTimeout(10 * time.Second).Should(BeNotFoundError())
 					}
 
 					By("Ensure Seed is gone")
 					Eventually(func() error {
 						return testClient.Get(ctx, client.ObjectKeyFromObject(seed), seed)
-					}).Should(BeNotFoundError())
+					}).WithTimeout(10 * time.Second).Should(BeNotFoundError())
```

Comment:
Without increasing the timeout, the test doesn't seem to pass against an actual cluster.

Reply:
Let's remove the hardcoded timeouts here that are required for running against an actual cluster.
When running against an actual cluster, you can overwrite the timeouts by setting `GOMEGA_DEFAULT_EVENTUALLY_TIMEOUT` as needed.